lib/crawly/engine.ex

```diff
@@ -33,58 +40,149 @@ defmodule Crawly.Engine do
     GenServer.call(__MODULE__, {:stop_spider, spider_name})
   end
 
-  @spec running_spiders() :: started_spiders()
+  @spec list_spiders() :: [spider_info()]
+  def list_spiders() do
+    GenServer.call(__MODULE__, :list_spiders)
+  end
+
+  @spec running_spiders() :: [spider_info()]
   def running_spiders() do
     GenServer.call(__MODULE__, :running_spiders)
   end
 
+  @spec get_spider(module()) :: spider_info()
```

Comment:
Should it be re-named to `get_spider_info` instead? What is the plan for this function? Do we want to get spider stats from it in the future?